File: app/src/main/java/com/hillal/hhhhhhh/ui/transactions/TransactionViewModel.java
package com.hillal.hhhhhhh.ui.transactions;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;
import com.hillal.hhhhhhh.data.model.Transaction;
import com.hillal.hhhhhhh.data.repository.AccountRepository;

import java.util.List;

public class TransactionViewModel extends ViewModel {
    private final AccountRepository accountRepository;

    public TransactionViewModel(AccountRepository accountRepository) {
        this.accountRepository = accountRepository;
    }

    public LiveData<List<Transaction>> getTransactionsForAccount(long accountId) {
        return accountRepository.getTransactionsForAccount(accountId);
    }

    public LiveData<Double> getAccountBalance(long accountId) {
        return accountRepository.getAccountBalance(accountId);
    }

    public void insertTransaction(Transaction transaction) {
        accountRepository.insertTransaction(transaction);
    }

    public void updateTransaction(Transaction transaction) {
        accountRepository.updateTransaction(transaction);
    }

    public void deleteTransaction(Transaction transaction) {
        accountRepository.deleteTransaction(transaction);
    }
} 

File: app/src/main/java/com/hillal/acc/ui/transactions/TransactionViewModel.kt
package com.hillal.acc.ui.transactions

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.hillal.acc.data.model.Transaction
import com.hillal.acc.data.repository.AccountRepository

class TransactionViewModel(private val accountRepository: AccountRepository) : ViewModel() {
    fun getTransactionsForAccount(accountId: Long): LiveData<MutableList<Transaction?>?>? {
        return accountRepository.getTransactionsForAccount(accountId)
    }

    fun getAccountBalance(accountId: Long): LiveData<Double?>? {
        return accountRepository.getAccountBalance(accountId)
    }

    fun getBalanceUntilTransaction(accountId: Long, transactionDate: Long, transactionId: Long, currency: String): LiveData<Double> {
        return accountRepository.database.transactionDao().getBalanceUntilTransaction(accountId, transactionDate, transactionId, currency)
    }

    fun insertTransaction(transaction: Transaction?) {
        accountRepository.insertTransaction(transaction)
    }

    fun updateTransaction(transaction: Transaction?) {
        accountRepository.updateTransaction(transaction)
    }

    fun deleteTransaction(transaction: Transaction?) {
        accountRepository.deleteTransaction(transaction)
    }
}